framework/src/base/FEProblem.C

```diff
@@ -2948,6 +3088,30 @@ FEProblem::setCouplingMatrix(CouplingMatrix * cm)
   _cm = cm;
 }
 
+void
+FEProblem::setNonlocalCouplingMatrix()
+{
+  unsigned int n_vars = _nl.nVariables();
+  _nonlocal_cm = new CouplingMatrix(n_vars);
```

Comment:
This `new` has no corresponding delete. I suggest using an `std::unique_ptr` to store this.